src/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+if(VMA_BUILD_EXAMPLE_APP_SHADERS)
+    add_subdirectory(Shaders)
+endif()
+
+if(VMA_BUILD_REPLAY_APP)
+    add_subdirectory(VmaReplay)
+endif()
+
+set(VMA_LIBRARY_SOURCE_FILES
+    VmaUsage.cpp
+)
+
+# Use folder structure as filters in project map
+foreach(FILE ${VMA_LIBRARY_SOURCE_FILES})
+    get_filename_component(PARENT_DIR "${FILE}" PATH)
+
+    string(REPLACE "/" "\\" GROUP "${PARENT_DIR}")
+
+    if("${FILE}" MATCHES ".*\\.cpp")
+        set(GROUP "Source Files\\${GROUP}")
+    elseif("${FILE}" MATCHES ".*\\.hpp")
+        set(GROUP "Header Files\\${GROUP}")
+    endif()
+
+    source_group("${GROUP}" FILES "${FILE}")
+endforeach()
+
+add_library(VulkanMemoryAllocator ${VMA_LIBRARY_SOURCE_FILES})
+
+set_target_properties(
+    VulkanMemoryAllocator PROPERTIES
+
+    CXX_EXTENSIONS OFF
+    # Use C++14
+    CXX_STANDARD 14
+    CXX_STANDARD_REQUIRED ON
+)
+
+target_include_directories(
+    VulkanMemoryAllocator
+
+    PUBLIC
+    ${PROJECT_SOURCE_DIR}/include
+)
+
+target_link_libraries(
+    VulkanMemoryAllocator
+
+    PUBLIC
+    Vulkan::Vulkan
+)
+
+set(VMA_EXAMPLE_SOURCE_FILES
+    Common.cpp
+    SparseBindingTest.cpp
+    Tests.cpp
+    VulkanSample.cpp
+)
+
+# Use folder structure as filters in project map
+foreach(FILE ${VMA_EXAMPLE_SOURCE_FILES})
+    get_filename_component(PARENT_DIR "${FILE}" PATH)
+
+    string(REPLACE "/" "\\" GROUP "${PARENT_DIR}")
+
+    if("${FILE}" MATCHES ".*\\.cpp")
+        set(GROUP "Source Files\\${GROUP}")
+    elseif("${FILE}" MATCHES ".*\\.hpp")
+        set(GROUP "Header Files\\${GROUP}")
+    endif()
+
+    source_group("${GROUP}" FILES "${FILE}")
+endforeach()
+
+if (VMA_BUILD_EXAMPLE_APP)
+    if(WIN32)
+        add_executable(VmaExample ${VMA_EXAMPLE_SOURCE_FILES})
+
+        # Visual Studio specific settings
+        if(${CMAKE_GENERATOR} MATCHES "Visual Studio.*")
+            # Use Unicode instead of multibyte set
+            add_definitions(-DUNICODE -D_UNICODE)
+            
+            # Set VmaExample as startup project
+            set_property(DIRECTORY ${PROJECT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT "VmaExample")
+            
+            # Enable multithreaded compiling
+            target_compile_options(VmaExample PRIVATE "/MP")
+
+            # Set working directory for Visual Studio debugger
+            set_target_properties(
+                VmaExample
+                PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/bin"
+            )
+        endif()
+
+        set_target_properties(
+            VmaExample PROPERTIES
+
+            CXX_EXTENSIONS OFF
+            # Use C++14
+            CXX_STANDARD 14
+            CXX_STANDARD_REQUIRED ON
+        )
+
+        target_link_libraries(
+            VmaExample
+
+            PRIVATE
+            VulkanMemoryAllocator
+            Vulkan::Vulkan
+        )
+    else()
+        message(STATUS "VmaExample application is not supported to Linux")
```

Comment:
Current state of the project is to compile the sample app, with code of the actual app disabled using `#ifdef _WIN32` inside source files, so that other platforms at least check if the library code compiles. As I understand, in your Cmake the VMA is still compiled because it is a separate LIB, so we won't check less than before by disabling the sample app?

Reply:
I added this because of the following reason: On Linux the code looks like this:

```
#else // #ifdef _WIN32

#include "VmaUsage.h"

int main()
{
}

#endif // #ifdef _WIN32
```

On some Linux platforms like my Ubuntu 20.04 compiling the VMA example application does **not** result in an executable file, but it generates a shared object instead:

```
file ./bin/VmaExample
./bin/VmaExample: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=ffa8949669582487287a59f98e8f472e0acf854e, for GNU/Linux 3.2.0, with debug_info, not stripped
```

I think the compiler sees that the application does nothing and prefers to make it a shared object then? I can't explain why this is the case, but it took me quite some time to understand what's wrong. I added this to inform other Linux devs. Some others tested it for me and on their system an executable (which does nothing) was obtained. I think in both cases it's better to inform the developers that the example application is not ported to Linux.